cloud/services/container/nodepools/reconcile.go

```diff
@@ -166,6 +166,19 @@ func (s *Service) Reconcile(ctx context.Context) (ctrl.Result, error) {
 		return ctrl.Result{RequeueAfter: reconciler.DefaultRetryTime}, nil
 	}
 
+	needUpdateNodePool, updateNodePoolRequest := s.checkDiffAndPrepareUpdateNodePool(nodePool)
```

Comment:
The `checkDiffAndPrepareUpdateVersionOrImage` block actually calls `UpdateNodePool` in the underlying logic. I think we can combine these so we're updating fields in one API call. Otherwise the second update call will block on the first. I'm doing something like this in my PR https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1026/files#diff-1d279184a2f22d9c46b9f844a1852f6a04956ef24533b4f066bc9e900abd71bdR147.